x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -377,12 +372,25 @@ export class AlertsClient {
       );
       const scheduledTask = await this.scheduleAlert(id, attributes.alertTypeId);
       await this.savedObjectsClient.update('alert', id, { scheduledTaskId: scheduledTask.id });
-      await this.invalidateApiKey({ apiKey: attributes.apiKey });
```

Comment:
It looks like we create an API when an alert is created, independent of the `enable` setting.  https://github.com/elastic/kibana/blob/d73e15f4147d1de9a7ee15517cbf6a90f4d4b213/x-pack/legacy/plugins/alerting/server/alerts_client.ts#L149-L160

I think we probably should *NOT* be doing that, as such keys would be invalidated here previously, but now they won't, so we're going to leave dangling valid api keys.  I think SIEM may create their alerts disabled, so we can check with them.  Suggest doing that as a separate issue, seems likely to affect other things (maybe mostly tests)

Reply:
Totally agree, it somehow creates API keys now regardless if they're enabled on alert creation. Will open an issue 👍 good catch!

Reply:
Created: #57036.